openhgnn/trainerflow/SeHGNN_trainer.py

```diff
@@ -0,0 +1,866 @@
+                end = time.time()
+
+                log = "Epoch {}, Time(s): {:.4f}, estimated train loss {:.4f}, acc {:.4f}\n".format(epoch, end-start, loss, acc*100)
+                torch.cuda.empty_cache()
+
+                if epoch % args.eval_every == 0:
+                    with torch.no_grad():
+                        model.eval()
+                        raw_preds = []
+
+                        start = time.time()
+                        for batch_feats, batch_label_feats, batch_labels_emb in eval_loader:
+                            batch_feats = {k: v.to(device) for k,v in batch_feats.items()}
+                            batch_label_feats = {k: v.to(device) for k,v in batch_label_feats.items()}
+                            batch_labels_emb = batch_labels_emb.to(device)
+                            raw_preds.append(model(batch_feats, batch_label_feats, batch_labels_emb).cpu())
+                        raw_preds = torch.cat(raw_preds, dim=0)
+
+                        loss_val = loss_fcn(raw_preds[:valid_node_nums], labels[trainval_point:valtest_point]).item()
+                        loss_test = loss_fcn(raw_preds[valid_node_nums:valid_node_nums+test_node_nums], labels[valtest_point:total_num_nodes]).item()
+
+                        preds = raw_preds.argmax(dim=-1)
+                        val_acc = evaluator(preds[:valid_node_nums], labels[trainval_point:valtest_point])
+                        test_acc = evaluator(preds[valid_node_nums:valid_node_nums+test_node_nums], labels[valtest_point:total_num_nodes])
+
+                        end = time.time()
+                        log += f'Time: {end-start}, Val loss: {loss_val}, Test loss: {loss_test}\n'
+                        log += 'Val acc: {:.4f}, Test acc: {:.4f}\n'.format(val_acc*100, test_acc*100)
+
+                    if val_acc > best_val_acc:
+                        best_epoch = epoch
+                        best_val_acc = val_acc
+                        best_test_acc = test_acc
+
+                        torch.save(model.state_dict(), f'{checkpt_file}_{stage}.pkl')
+                        count = 0
+                    else:
+                        count = count + args.eval_every
+                        if count >= args.patience:
+                            break
+                    log += "Best Epoch {},Val {:.4f}, Test {:.4f}".format(best_epoch, best_val_acc*100, best_test_acc*100)
+                print(log, flush=True)
+
+            print("Best Epoch {}, Val {:.4f}, Test {:.4f}".format(best_epoch, best_val_acc*100, best_test_acc*100))
+
+            model.load_state_dict(torch.load(checkpt_file+f'_{stage}.pkl'))
+            raw_preds = self.gen_output_torch(model, feats, label_feats, label_emb, all_loader, device)
+            torch.save(raw_preds, checkpt_file+f'_{stage}.pt')
+
+
+    def parse_args(self, args=None):
```

Comment:
This function should be removed. We already use the ``conifg.py`` and ``config.ini`` to get the arguments, so why do you need another parser?